From f1b62a36fd92170e36d40219998837d9822d3c9f Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss@m4x.org>
Date: Sun, 7 Sep 2014 23:32:21 +0200
Subject: [PATCH] Allow loadkeys to read usr_t files

loadeys loads a keymap from files in /usr.
Debian console-data package installs keymaps in /usr/share/keymaps [1].
Arch Linux kbd package installs keymaps in /usr/share/kbd/keymaps [2].

[1] https://packages.debian.org/sid/all/console-data/filelist
[2] https://www.archlinux.org/packages/core/x86_64/kbd/files/
---
 policy/modules/contrib/loadkeys.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/contrib/loadkeys.te b/policy/modules/contrib/loadkeys.te
index d2f4643..72ba702 100644
--- a/policy/modules/contrib/loadkeys.te
+++ b/policy/modules/contrib/loadkeys.te
@@ -27,6 +27,8 @@ corecmd_exec_shell(loadkeys_t)
 
 files_read_etc_files(loadkeys_t)
 files_read_etc_runtime_files(loadkeys_t)
+# keymap files are in /usr/share/keymaps or /usr/share/kbd/keymaps
+files_read_usr_files(loadkeys_t)
 
 term_dontaudit_use_console(loadkeys_t)
 term_use_unallocated_ttys(loadkeys_t)
-- 
2.5.0

